### DISCUSSION TITLE
debugging database query not working

### DISCUSSION THREAD

**Discussion by vikrambiwal** [adonisjs/core#2818] [Help] [answered]
I have set `debug: true,` in `config/database.ts` but the query does not print in the console.

`Database.query().select('*').debug(true)` also not owrking.

**Comment by avidcoder123** [comment]
Try listening to the `db:query` event and log the query
`Event.on("db:query", console.log)` will log every query

**Comment by RomainLanz** [comment, marked as answer]
Hey @vikrambiwal! 👋 

Please, re-read the documentation on how to debug your queries.
You need to add en event listener if you want them printed to the console.

📚 https://docs.adonisjs.com/guides/database/debugging#listening-to-the-event
